kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -370,17 +370,44 @@ impl<'tcx> GotocCtx<'tcx> {
             }};
         }
 
-        if let Some(_stripped) = intrinsic.strip_prefix("simd_shuffle") {
-            // TODO: can be empty now (i.e. `simd_shuffle` instead of `simd_shuffle8`)
-            // `parse` fails on empty, so comment that bit of code out.
-            // To re-enable this we'll need to investigate how size is computed now.
-            // let n: u64 = stripped.parse().unwrap();
-            return unstable_codegen!(self.codegen_intrinsic_simd_shuffle(
-                fargs,
-                p,
-                cbmc_ret_ty,
-                n
-            ));
+        if let Some(stripped) = intrinsic.strip_prefix("simd_shuffle") {
```

Comment:
nit: can you move this to its own function? Maybe call that from inside the codegen intrinsic function.

I think this function is already pretty big, so i would prefer not adding more logic in here.

Reply:
Agreed, moved into a method (since we need to span errors). Thanks!